docs: add getting started tutorial for dotnet

CaptainStandby opened the pull request:
## Related Issue or Design Document

closes #830

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

alnr commented on docs/getting-started/integrate-auth/dotnet.mdx:
```diff
@@ -0,0 +1,277 @@
+
+### Usage of Ory Tunnel
+
+For this example we will be using the [Ory Tunnel](../../guides/cli/proxy-and-tunnel#ory-tunnel). The interactions beeing
```
> a
```suggestion
We will be using the [Ory Tunnel](../../guides/cli/proxy-and-tunnel#ory-tunnel) in this tutorial.
```

alnr commented on docs/getting-started/integrate-auth/dotnet.mdx:
```diff
@@ -0,0 +1,277 @@
+Next, create middleware that checks if the user has a valid session. The session is considered valid when the user is
+authenticated.
+
+The middleware passes the current `HttpContext` cookies to the Ory client to check if the session is valid.
```
```suggestion
The middleware passes the request cookies to the Ory client to check if the session is valid.
If the session is valid and active, the session information is recorded in `ctx`.
```

alnr commented on docs/getting-started/integrate-auth/dotnet.mdx:
```diff
@@ -0,0 +1,277 @@
+
+With all of the pieces in place, it's time to test your application. Follow these steps:
+
+1. Start your ASP.NET app:
```
I followed the tutorial exactly to this point and get this:

```
$ dotnet run
Building...
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(9,15): error CS0246: The type or namespace name 'FrontendApi' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(9,31): error CS0246: The type or namespace name 'Configuration' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(35,9): error CS0246: The type or namespace name 'ApiException' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]

The build failed. Fix the build errors and run again.
```

CaptainStandby replied:
I have left out the namespace imports in the code boxes on purpose to remove noise. Maybe they should be added to make the tutorial truly self contained?

It's only two lines of Code, so I will add the imports.

aeneasr commented:
Very nice :) To ensure that the example always works, can you please add it to the end-to-end tests? :)

https://github.com/ory/docs/blob/master/tests/playwright/protect-page-login.spec.ts#L21

alnr commented:
I've worked through the tutorial again and got it working in principle. But the logged-in page says `Hello World` instead of the session info 🤷 

In general, I would strongly recommend that the last two listings of `Program.cs` be copy-pasteable for those lazy fellas like yours truly.

Also: is there a better way to fix the dotnet default HTTP port? That tripped me up while working through the tutorial, especially since it's needed to open the browser to the correct page *and* correctly configure the `ory tunnel` command. Optional.

Final nit: The tutorial mentions `ORY_SDK_URL` but the CLI says `--project <slug>` is preferred. Maybe that could be improved. Optional.

CaptainStandby commented:
@alnr 

> I've worked through the tutorial again and got it working in principle. But the logged-in page says `Hello World` instead of the session info 🤷

I have updated the code blocks to emphasize that `app.MapGet("/", () => "Hello World!");` needs to be removed.

> In general, I would strongly recommend that the last two listings of `Program.cs` be copy-pasteable for those lazy fellas like yours truly.

They should be copyable for the most part.
I feel that the "diff" style used here gives a better understanding of the different parts and their purpose, so I would like to keep it this way. The actual source code is linked at the top, so there is a reference that is completely copy-pasteable.

> Also: is there a better way to fix the dotnet default HTTP port? That tripped me up while working through the tutorial, especially since it's needed to open the browser to the correct page _and_ correctly configure the `ory tunnel` command. Optional.

I have changed some things to make this explicit and part of the tutorial.

> Final nit: The tutorial mentions `ORY_SDK_URL` but the CLI says `--project <slug>` is preferred. Maybe that could be improved. Optional.

Switched to using the slug

CaptainStandby commented:
> Very nice :) To ensure that the example always works, can you please add it to the end-to-end tests? :)
> 
> https://github.com/ory/docs/blob/master/tests/playwright/protect-page-login.spec.ts#L21

done :-)

alnr commented:
Sorry for being anal in this review! 😅 

1. can you move the port override (thank you!) in the first listing of Program.cs? 
2. can you please choose a different default port than 5000? It's an issue on default settings on macOS 🙄 https://developer.apple.com/forums/thread/682332

alnr reviewed (approved):
Perfect!
